src/test/java/seedu/address/logic/commands/AddMaintainerCommandTest.java

```diff
@@ -0,0 +1,207 @@
+import seedu.address.model.person.Maintainer;
+import seedu.address.model.person.Person;
+import seedu.address.testutil.MaintainerBuilder;
+
+public class AddMaintainerCommandTest {
+
+    @Test
+    public void constructor_nullPerson_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new AddMaintainerCommand(null));
+    }
+
+    @Test
+    public void execute_personAcceptedByModel_addSuccessful() throws Exception {
+        ModelStubAcceptingPersonAdded modelStub = new ModelStubAcceptingPersonAdded();
+        Maintainer validPerson = new MaintainerBuilder().build();
+
+        CommandResult commandResult = new AddMaintainerCommand(validPerson).execute(modelStub);
+
+        assertEquals(String.format(AddMaintainerCommand.MESSAGE_SUCCESS, Messages.format(validPerson)),
+                commandResult.getFeedbackToUser());
+        assertEquals(Arrays.asList(validPerson), modelStub.personsAdded);
+    }
+
+    @Test
+    public void execute_duplicatePerson_throwsCommandException() {
+        Maintainer validPerson = new MaintainerBuilder().build();
+        AddMaintainerCommand addCommand = new AddMaintainerCommand(validPerson);
+        ModelStub modelStub = new ModelStubWithPerson(validPerson);
+
+        assertThrows(CommandException.class,
+                AddMaintainerCommand.MESSAGE_DUPLICATE_PERSON, () -> addCommand.execute(modelStub));
+    }
+
+    @Test
+    public void equals() {
+        Maintainer alice = new MaintainerBuilder().withName("Alice").build();
+        Maintainer bob = new MaintainerBuilder().withName("Bob").build();
+        AddMaintainerCommand addAliceCommand = new AddMaintainerCommand(alice);
+        AddMaintainerCommand addBobCommand = new AddMaintainerCommand(bob);
+
+        // same object -> returns true
+        assertTrue(addAliceCommand.equals(addAliceCommand));
+
+        // same values -> returns true
+        AddMaintainerCommand addAliceCommandCopy = new AddMaintainerCommand(alice);
+        assertTrue(addAliceCommand.equals(addAliceCommandCopy));
+
+        // different types -> returns false
+        assertFalse(addAliceCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(addAliceCommand.equals(null));
+
+        // different person -> returns false
+        assertFalse(addAliceCommand.equals(addBobCommand));
+    }
+
+    @Test
+    public void toStringMethod() {
+        AddMaintainerCommand addCommand = new AddMaintainerCommand(ALICEMAINTAINER);
+        String expected = AddMaintainerCommand.class.getCanonicalName()
+                + "{toAdd=" + ALICEMAINTAINER + "}";
+        assertEquals(expected, addCommand.toString());
+    }
+
+    /**
+     * A default model stub that have all of the methods failing.
+     */
+    private class ModelStub implements Model {
+        @Override
+        public void setUserPrefs(ReadOnlyUserPrefs userPrefs) {
+            throw new AssertionError("This method should not be called.");
```

Comment:
Maybe abstract out the error message into a constant variable since it is repeated several times.